.github/workflows/publish-nx.yml

```diff
@@ -1,38 +1,42 @@
-name: CI
+name: Publish NX
 on:
   workflow_dispatch:
   push:
     branches:
-      # Change this if your primary branch is not main
       - main
     paths:
       - CHANGELOG.md
 
 jobs:
   main:
     runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      id-token: write
+      packages: write
```

Comment:
Is this required? I don't think we are using github packages.
https://docs.github.com/en/packages/learn-github-packages/about-permissions-for-github-packages#visibility-and-access-permissions-for-packages
https://docs.github.com/en/packages/learn-github-packages/about-permissions-for-github-packages#access-tokens

If we are, is read on `packages` enough?

Reply:
Ah, probably not. This is probably just an artifact of my "solution shotgun" fixing.

Reply:
According to https://docs.github.com/en/actions/publishing-packages/publishing-nodejs-packages#publishing-packages-to-the-npm-registry your original config should work. 
I note this is using v4 of the action though which could be using a newer version of npm in some part. Npm login defaults were changed were made between node 16 and node 20 .

https://github.com/actions/setup-node/compare/v3...v4#diff-1243c5424efaaa19bd8e813c5e6f6da46316e63761421b3e5f5c8ced9a36e6b6

https://docs.npmjs.com/accessing-npm-using-2fa#sign-in-from-the-command-line-using---auth-typeweb

Reply:
My original do you mean, the base? It definitely doesn't work. I'll try v4

Reply:
v4 Doesn't allow me to remove "Configure npm" step:
https://github.com/telicent-oss/rdf-libraries/actions/runs/8139202853


I'll remove the `packages: write` in a new PR, and then I'll leave this for now as this workflow works, and I need to check the other 6 or so workflows